src/App.tsx

```diff
@@ -1,14 +1,58 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [visiableTodos, setVisiableTodos] = useState<Todo[]>([]);
+  const [loader, setLoader] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filter, setFilter] = useState('all');
+  const [query, setQuery] = useState('');
+
+  const filterByStatus = (status: string, todosList: Todo[]) => {
+    switch (status) {
+      case 'active':
+        setVisiableTodos(todosList.filter((todo => todo.completed)));
+        break;
+
+      case 'completed':
+        setVisiableTodos(todosList.filter((todo => !todo.completed)));
+        break;
+
+      case 'all':
+      default:
+        setVisiableTodos(todosList);
+        break;
+    }
+  };
```

Comment:
to make it even better let's either wrap this in useCallback or move outside of the component to prevent recreating function on each rerender.